src/views/Clients/ViewClients/index.jsx

```diff
@@ -82,11 +82,16 @@ export default class ViewWorker extends PureComponent {
     this.setState({ clientSearch: target.value });
   };
 
-  handleClientSearchSubmit = e => {
+  handleClientSearchSubmit = async e => {
     if (e) {
       e.preventDefault();
     }
 
+    await this.setState({
+      clientSearch: this.state.clientSearch.replace(/\s/g, ''),
+    });
+    console.log(`${this.state.clientSearch}test`)
+
```

Comment:
No need to modify the state. When refetching you can simply remove trailing spaces from both ends using the `trim` method:

```jsx
clientOptions: {
  ...(clientSearch ? { prefix: clientSearch.trim() } : null),
},
``